### PR TITLE
Workflow reference

### PR DESCRIPTION
**Pull request by perolavsvendsen**
Solving #56

### CODE REVIEW

File: src/fmu/dataio/_export_item.py [hunk 1]
```diff
@@ -824,10 +823,6 @@ def _data_process_object_arrowtable(self):
         meta["bbox"] = None
         logger.info("Process data metadata for ArrowTable... done!!")
 
-    def _fmu_inject_workflow(self):
-        """Inject workflow into fmu metadata block."""
-        self.dataio.metadata4fmu["workflow"] = self.dataio.workflow
```
Comment (jcrivenaes) [inline src/fmu/dataio/_export_item.py:829]: Where is `_fmu_inject_workflow` now, or it it not needed any more?

Comment (perolavsvendsen) [reply src/fmu/dataio/_export_item.py:829]: I don't think it is needed (?) As far as I could see, the only function was to insert `self.dataio.workflow` into the fmu-block. But workflow was already there, inserted as part of the `_get_fmu_meta`. Also, while the `workflow` tag was initially inserted correctly (as an object), this function (when called later) overwrote the workflow object with the string.

(Confirmation/second opinion on this welcomed - I may have missed something here)

**Comment by perolavsvendsen** [comment]
Currently, the workflow argument is expecting a string. However, it has been given by many as a dictionary (this is why it has validated before). I considered to make it accept both (and sort it out under the hood), but I believe it's bad practice.

Also, it may be confusing that the `workflow` argument is used for the `workflow.reference` in outgoing metadata. I did consider to change the argument to `workflow_reference` but this would involve an API change.

@jcrivenaes

**Comment by perolavsvendsen** [comment]
Offline discussion: Allow for `workflow` to be both `str` and `dict`. Similar to current method for `content` (inherit code if possible).

If `str`, put the input into `workflow.reference`. If `dict`, check that `reference` is there and is a string, then put the input in the `workflow` tag.

File: src/fmu/dataio/dataio.py [hunk 2]
```diff
@@ -409,17 +409,17 @@ def _get_meta_fmu(self) -> None:
         The fmu block consist of these subkeys:
             model:
             case:
-            workflow:
+            workflow: Can be dict or str. If str, it is to be used as the value for
+                      the 'reference' key and be inserted into dict. If dict, it shall
+                      contain the 'reference' key then be used directly.
             element:  # if aggadation
```
Comment (jcrivenaes) [inline src/fmu/dataio/dataio.py:415]: Some grammar issues here

**Review by jcrivenaes — APPROVED** [review]
LGTM

**Pull request merged**